src/main/java/bio/terra/pdao/bigquery/BigQueryPdao.java

```diff
@@ -564,14 +564,21 @@ private String prefixName(String name) {
 
     private Schema buildSchema(Table table, boolean addRowIdColumn) {
         List<Field> fieldList = new ArrayList<>();
+        List<String> primaryKeys = table.getPrimaryKey().orElse(Collections.emptyList())
+            .stream()
+            .map(Column::getName)
+            .collect(Collectors.toList());
 
         if (addRowIdColumn) {
             fieldList.add(Field.of(PDAO_ROW_ID_COLUMN, LegacySQLTypeName.STRING));
         }
 
         for (Column column : table.getColumns()) {
+            boolean isPk = primaryKeys.contains(column.getName());
```

Comment:
nit: conditional operator is only useful when it helps readability. I don't think it does here.